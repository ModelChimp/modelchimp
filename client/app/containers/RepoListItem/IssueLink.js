import styled from 'styled-components';
import NormalA from 'components/A';

const IssueLink = styled(NormalA)`
  height: 100%;
  color: black;
  display: flex;
  align-items: center;
  justify-content: center;
`;

export default IssueLink;
