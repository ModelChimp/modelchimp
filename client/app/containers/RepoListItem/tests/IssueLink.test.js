import React from 'react';
import { mount } from 'enzyme';
import { enzymeFind } from 'styled-components/test-utils';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import IssueLink from '../IssueLink';

describe('<IssueLink />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(<IssueLink />).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should have a className attribute', () => {
    const wrapper = mount(<IssueLink />);
    const renderedComponent = enzymeFind(wrapper, IssueLink);
    expect(renderedComponent.prop('className')).toBeDefined();
  });

  it('should adopt a valid attribute', () => {
    const id = 'test';
    const wrapper = mount(<IssueLink id={id} />);
    const renderedComponent = enzymeFind(wrapper, IssueLink);
    expect(renderedComponent.prop('id')).toEqual(id);
  });

  it('should not adopt an invalid attribute', () => {
    const wrapper = mount(<IssueLink attribute="test" />);
    const renderedComponent = enzymeFind(wrapper, IssueLink);
    expect(renderedComponent.prop('attribute')).toBeUndefined();
  });
});
