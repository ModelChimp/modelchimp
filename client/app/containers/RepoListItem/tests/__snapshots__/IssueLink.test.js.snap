// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IssueLink /> should match the snapshot 1`] = `
.c0 {
  color: #41addd;
  height: 100%;
  color: black;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0:hover {
  color: #6cc0e5;
}

<a
  className="c0"
/>
`;
